### PR TITLE
Add startpoint_method to Problem

### LINKED ISSUE #1035: Add startpoint_method to Problem
It's not intuitive that `x_guesses` is part of `Problem`, but `startpoint_method`s are handled separately. See also discussion in #1017. Would be great to change that.

### PR DESCRIPTION
**Pull request by dweindl**
It's not intuitive that `x_guesses` is part of `Problem`, but `startpoint_method`s are handled separately. See also discussion in #1017.

This patch
* adds `startpoint_method` to `Problem`
* during PEtab import, sets `Problem.startpoint_method` based on the PEtab problem

To be discussed: Do we want to keep the existing `startpoint_method` argument in the long term or should it be removed (deprecated)?

Closes #1035

### CODE REVIEW

**Review by yannikschaelte — APPROVED** [review]
looks reasonable to me! I'd suggest deprecating the minimize.startpoint_method argument then.

**Comment by dweindl** [comment]
On hold. Requires `dill>0.3.6` (current `master` works), otherwise the dynesty notebook fails at `sampler.save_internal_sampler("dynesty.dill")` with `TypeError: cannot pickle '_abc._abc_data' object`.

File: pypesto/problem/base.py [hunk 1]
```diff
@@ -90,6 +99,7 @@ def __init__(
         lb_init: Union[np.ndarray, List[float], None] = None,
         ub_init: Union[np.ndarray, List[float], None] = None,
         copy_objective: bool = True,
+        startpoint_method: Union[StartpointMethod, Callable, bool] = None,
```
Comment (dilpath) [inline pypesto/problem/base.py:102]: doc

**Review by dilpath — APPROVED** [review]
:+1:

**Review by FFroehlich — APPROVED** [review]
👍

**Pull request merged**